includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -153,15 +163,34 @@ protected function update_post_meta( &$order ) {
 			'_customer_ip_address'  => 'customer_ip_address',
 			'_customer_user_agent'  => 'customer_user_agent',
 			'_created_via'          => 'created_via',
-			'_completed_date'       => 'date_completed',
-			'_paid_date'            => 'date_paid',
+			'_date_completed'       => 'date_completed',
+			'_date_paid'            => 'date_paid',
 			'_cart_hash'            => 'cart_hash',
 		);
 
 		$props_to_update = $this->get_props_to_update( $order, $meta_key_to_props );
+
 		foreach ( $props_to_update as $meta_key => $prop ) {
 			$value = $order->{"get_$prop"}( 'edit' );
-			update_post_meta( $id, $meta_key, $value );
+
+			if ( 'date_paid' === $prop ) {
+				// In 2.7.x we store this as a UTC timestamp.
+				update_post_meta( $id, $meta_key, ! is_null( $value ) ? $value->getTimestamp() : '', true );
```

Comment:
Passing the `prev_value` to `update_post_meta` will not allow updates to these meta fields.